enhancements/installer/single-node-installation-bootstrap-in-place.md

```diff
@@ -0,0 +1,546 @@
+2. DNS records:
+* api.<cluster_name>.<base_domain>.
+* api-int.<cluster_name>.<base_domain>.
+* *.apps.<cluster_name>.<base_domain>.
+* <hostname>.<cluster_name>.<base_domain>.
+### Initial Proof-of-Concept
+
+A proof-of-concept implementation is available for experimenting with
+the design.
+
+To try it out: [bootstrap-in-place-poc](https://github.com/eranco74/bootstrap-in-place-poc.git)
+
+### Risks and Mitigations
+
+*What are the risks of this proposal and how do we mitigate. Think broadly. For
+example, consider both security and how this will impact the larger OKD
+ecosystem.*
+
+*How will security be reviewed and by whom? How will UX be reviewed and by whom?*
+
+#### Custom Manifests for CRDs
+
+One limitation of single-node deployments not present in multi-node
+clusters is handling some custom resource definitions (CRDs). During
+bootstrapping of a multi-node cluster, the bootstrap host and real
+cluster hosts run in parallel for a time. This means that the
+bootstrap host can iterate publishing manifests to the API server
+until the operators running on the other hosts are up and define their
+CRDs. If it takes a little while for those operators to install their
+CRDs, the bootstrap host can wait and retry the operation. In a
+single-node deployment, the bootstrap environment and real node are
+not active at the same time. This means the bootstrap process may
+block if it tries to create a custom resource using a CRD that is not
+installed.
+
+While most CRDs are created by the `cluster-version-operator`, some
+CRDs are created later by the cluster operators. These CRDs from
+cluster operators are not present during bootstrapping:
+
+* clusternetworks.network.openshift.io
+* controllerconfigs.machineconfiguration.openshift.io
+* egressnetworkpolicies.network.openshift.io
+* hostsubnets.network.openshift.io
+* ippools.whereabouts.cni.cncf.io
+* netnamespaces.network.openshift.io
+* network-attachment-definitions.k8s.cni.cncf.io
+* overlappingrangeipreservations.whereabouts.cni.cncf.io
+* volumesnapshotclasses.snapshot.storage.k8s.io
+* volumesnapshotcontents.snapshot.storage.k8s.io
+* volumesnapshots.snapshot.storage.k8s.io
+
+This limitation is unlikely to be triggered by manifests created by
+the OpenShift installer, but we cannot control what extra manifests
+users add to their deployment. Users need to be made aware of this
+limitation and encouraged to avoid creating custom manifests using
+CRDs installed by cluster operators instead of the
+`cluster-version-operator`.
+
+#### Post-reboot service makes single-node deployments different from multi-node clusters
+
+The `post-reboot` service is only used with single-node
+deployments. This makes those deployments look different in a way that
+may lead to confusion when debugging issues on the cluster. To
+mitigate this, we can add documentation to the troubleshooting guide
+to explain the service and its role in the cluster.
```

Comment:
Does the post-reboot service delete itself on the way out?

Reply:
yes,  the service will delete itself